### PR TITLE
[#1534] Fix test for accept encoding

### PR DESCRIPTION
**Pull request by funkybob**

### CODE REVIEW

**Comment by unbit** [comment]
Thanks, @xrmx safe for backport

**Comment by gdamjan** [comment]
Looking at the `http_add_uwsgi_header` function, before the line that this commit changes, there are other uses of headers with underscores and with `uwsgi_strncmp()`.

The code is a bit hard to follow, but in `http_headers_parse`, http_headers_parse is used after something that's called http_header_to_cgi that normalizes the header names.
